pkg/apis/core/validation/validation.go

```diff
@@ -4162,28 +4171,74 @@ func validateConfigMapNodeConfigSourceSpec(source *core.ConfigMapNodeConfigSourc
 	return append(allErrs, validateConfigMapNodeConfigSource(source, fldPath)...)
 }
 
+// validation specififc to Node.Status.Config
+func validateNodeConfigStatus(status *core.NodeConfigStatus, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if status.Assigned != nil {
+		// Assigned is a reflection of the spec, which acts as an ACK from the Kubelet regarding recognition,
+		// of the assignment. As such, it is validated against the same rules as the spec, rather than status.
```

Comment:
I thought you wanted resource version and uid reflected in the status, to know whether the node had observed a particular version of the source

Reply:
I suppose one possibility is we've seen the assigned reference and are getting an error fetching it, in which case we wouldn't have a uid or resourceVersion

Reply:
> I thought you wanted resource version and uid reflected in the status, to know whether the node had observed a particular version of the source

For `Active` and `LastKnownGood`, yes.

`Assigned` is just an ack, so you know what the Kubelet thinks you told it to do. When we start responding to ConfigMap mutations, the meaning of `ConfigSource`, assigned in spec, will become "I'm assigning you the latest config at this namespace/name, invariant to UID and ResourceVersion," and I want the meaning of `Assigned` in status to be "I recognize that I should use the latest config at that namespace/name, invariant to UID and ResourceVersion."

> I suppose one possibility is we've seen the assigned reference and are getting an error fetching it, in which case we wouldn't have a uid or resourceVersion

Exactly, `Assigned` gets acknowledged before the Kubelet tries to download the config, and if an error is encountered while downloading, `Error` gets set subsequently.